pkg/agent/baker.go

```diff
@@ -260,6 +260,12 @@ func getContainerServiceFuncMap(cs *api.ContainerService, profile *api.AgentPool
 		"IsKubernetes": func() bool {
 			return cs.Properties.OrchestratorProfile.IsKubernetes()
 		},
+		"FQDN": func() string {
+			if cs.Properties.HostedMasterProfile == nil {
+				return ""
+			}
+			return cs.Properties.HostedMasterProfile.FQDN
+		},
```

Comment:
why cannot we use GetParameter kubernetesEndpoint?

Reply:
couldn't make "GetParameter" work inside a separate script. 
Seems `getBase64EncodedGzippedCustomScript` will be evaluated before generate template. And it only uses parameter from `getContainerServiceFuncMap`.

Reply:
I see. In that case can you 
1. rename this one to  GetKubernetesEndpoint 
2. update the "{{ GetParameter kubernetesEndpoint }} " call to GetKubernetesEndpoint  in all places
3. remove the lines where we set the kubernetes parameter? In pkg/agent/params.go, L361

Reply:
fixed

Reply:
thanks!